File: examples/gateway-client.pl
#!/usr/bin/env perl
use strict;
use warnings;
use HTTP::Tiny;

# OpenClaw Gateway Client (Perl)
sub check_gateway {
    my ($base_url) = @_;
    my $response = HTTP::Tiny->new( timeout => 5 )->get("$base_url/health");
    return $response->{status};
}

my $url = $ARGV[0] // 'http://localhost:8080';
printf "Gateway %s -> HTTP %s\n", $url, check_gateway($url);

File: examples/gateway-client.php
<?php
// OpenClaw Gateway Client (PHP)

function checkGateway(string $baseUrl, string $endpoint = '/health'): int
{
    $ch = curl_init($baseUrl . $endpoint);
    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_NOBODY => true,
        CURLOPT_TIMEOUT => 5,
    ]);
    curl_exec($ch);
    $status = (int) curl_getinfo($ch, CURLINFO_RESPONSE_CODE);
    curl_close($ch);
    return $status;
}

$url = $argv[1] ?? 'http://localhost:8080';
printf("Gateway %s -> HTTP %d\n", $url, checkGateway($url));
